app/cdap/components/RecipeManagement/CreateRecipe/index.tsx

```diff
@@ -0,0 +1,168 @@
+
+const noErrorState = {
+  isRecipeNameError: false,
+  recipeNameErrorMessage: '',
+};
+
+export default function CreateRecipe({ setShowRecipeForm, setSnackbar }: ICreateRecipeProps) {
+  const [recipeFormData, setRecipeFormData] = useState<IRecipeFormData>({
+    recipeName: '',
+    description: '',
+    directives: [],
+  });
+
+  const [isSaveDisabled, setIsSaveDisabled] = useState<boolean>(true);
+  const [recipeNameErrorData, setRecipeNameErrorData] = useState(noErrorState);
+
+  // This static data has to be removed when we have actual API data, then directly we will get that data from store as directives
+  const recipeSteps = ['uppercase: body1', 'titlecase: body2'];
+
+  useEffect(() => {
+    handleSaveButtonMode();
+  }, [recipeNameErrorData.isRecipeNameError]);
+
+  const handleSaveButtonMode = (formData = recipeFormData) => {
+    if (
+      formData.recipeName === '' ||
+      formData.description === '' ||
+      formData.recipeName?.trim().length === 0 ||
+      formData.description?.trim().length === 0 ||
+      recipeNameErrorData.isRecipeNameError
+    ) {
+      setIsSaveDisabled(true);
+    } else {
+      setIsSaveDisabled(false);
+    }
+  };
+
+  const handleRecipeFormData = (formData) => {
+    setRecipeFormData(formData);
+    handleSaveButtonMode(formData);
+  };
+
+  const onRecipeNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    handleRecipeFormData({
+      ...recipeFormData,
+      recipeName: event.target.value,
+    });
+    validateRecipeNameExists.current(event.target.value);
+  };
+
+  const onRecipeDescriptionChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    handleRecipeFormData({
+      ...recipeFormData,
+      description: event.target.value,
+    });
+  };
+
+  const onFormSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    onRecipeDataSave(recipeFormData);
+  };
+
+  const onRecipeDataSave = (recipeFormData: IRecipeFormData) => {
+    const requestBody = {
+      recipeName: recipeFormData.recipeName,
+      description: recipeFormData.description,
+      directives: recipeSteps,
+    };
+    createRecipe(requestBody, onCreateRecipeResponse, onCreateRecipeError);
+  };
+
+  const onCreateRecipeResponse = () => {
+    setShowRecipeForm(false);
+    setSnackbar({
+      open: true,
+      isSuccess: true,
+      message: `${recipeSteps.length} ${T.translate(`${PREFIX}.recipeSaveSuccessMessage`)}`,
+    });
+  };
+
+  const onCreateRecipeError = (err) => {
+    setShowRecipeForm(false);
+    setSnackbar({
+      open: true,
+      isSuccess: false,
+      message: err.response.message,
+    });
+  };
+
+  const onCancel = () => {
+    setShowRecipeForm(false);
+  };
+
+  const onGetRecipeByNameError = (err, recipeName) => {
+    if (
+      err.statusCode === 404 &&
+      err.message === `recipe with name '${recipeName}' does not exist`
+    ) {
+      setRecipeNameErrorData(noErrorState);
+    }
+  };
```

Comment:
Check seems fragile. Needs discussion.

Reply:
+1

Reply:
Checked with @vanathi-g , the 404 check should be sufficient here.

Reply:
removed message check